src/continuous_agg.c

```diff
@@ -529,26 +541,33 @@ drop_continuous_agg(ContinuousAgg *agg, bool drop_user_view)
 
 		/* delete all related rows */
 		if (!raw_hypertable_has_other_caggs)
+		{
 			hypertable_invalidation_log_delete(form->raw_hypertable_id);
-
-		if (!raw_hypertable_has_other_caggs)
 			invalidation_threshold_delete(form->raw_hypertable_id);
+		}
+
 		materialization_invalidation_log_delete(form->mat_hypertable_id);
-		count++;
 
 		if (should_free)
 			heap_freetuple(tuple);
 	}
-	Assert(count == 1);
 
-	if (OidIsValid(rawht_trig.objectId))
-		ts_hypertable_drop_trigger(raw_hypertable, CAGGINVAL_TRIGGER_NAME);
+	/* Perform actual deletions now */
+	if (OidIsValid(user_view.objectId))
+		performDeletion(&user_view, DROP_RESTRICT, 0);
 
-	/* delete the materialization table */
-	ts_hypertable_drop(mat_hypertable, DROP_CASCADE);
+	if (OidIsValid(raw_hypertable_trig.objectId))
+		ts_hypertable_drop_trigger(raw_hypertable.objectId, CAGGINVAL_TRIGGER_NAME);
+
+	if (OidIsValid(mat_hypertable.objectId))
+	{
+		performDeletion(&mat_hypertable, DROP_CASCADE, 0);
```

Comment:
Why does this need CASCADE, while others don't? It would be good to at least explain this.

Reply:
I don't have an explanation for that, rather then it was there before. I can only imaging some user-case depending on this object

Reply:
CASCADE is required there, it will not work with RESTRICT. Likely for the hypertable it is expected to have depenent objects? (chunks, etc)

Reply:
I don't think cascade is required normally when dropping a hypertable. This looks potentially   dangerous, but I guess it is mitigated by this being an internal table.